.github/workflows/ci.yml

```diff
@@ -0,0 +1,100 @@
+# GitHub Actions CI definition for TensorBoard.
+#
+# YAML schema for GitHub Actions:
+# https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions
+#
+# Helpful YAML parser to clarify YAML syntax:
+# https://yaml-online-parser.appspot.com/
+
+# For now, we only use GitHub Actions for lint checks, pending better
+# support for hermetic-style caching. See:
+# https://github.com/actions/cache/issues/109
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - '[0-9]+.*'
+      - 'ci-*'
+  pull_request: {}
+
+env:
+  BUILDIFIER: '0.29.0'
+  BUILDIFIER_SHA256SUM: '4c985c883eafdde9c0e8cf3c8595b8bfdf32e77571c369bf8ddae83b042028d6'
+
+jobs:
+  lint-flake8:
+    runs-on: ubuntu-16.04
+    strategy:
+      fail-fast: false
+      matrix:
+        # flake8 should run on each Python version that we target,
+        # because the errors and warnings can differ due to language
+        # changes, and we want to catch them all.
+        python_version: ['3.6']
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python_version }}
+          architecture: 'x64'
+      - name: 'Install flake8'
+        run: |
+          python -m pip install -U pip
+          pip install -r ./tensorboard/pip_package/requirements_dev.txt
+      - run: pip freeze --all
+      - name: 'Lint Python code for errors with flake8'
+        # See: http://flake8.pycqa.org/en/3.7.8/user/error-codes.html
+        # Use the comment '# noqa: <error code>' to suppress.
+        run: flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+
+  lint-black-and-yaml:
```

Comment:
It's slightly odd to me to combine these together since they're pretty much linting disjoint sets of files that aren't particularly related.  I guess it's because they both involve a pip install, but unlike flake8 they don't need to run for every python version we target?

In practice since we are only running flake8 against 3.6 anyway it seems maybe simpler to just have a `lint-python-and-yaml` option, or to break it down to be even more granular and have `lint-black` and `lint-yaml` (or `lint-docs`) separately.